src/Compilers/CSharp/Test/Semantic/Semantics/AnonymousFunctionTests.cs

```diff
@@ -0,0 +1,739 @@
+            CreateCompilation(source).VerifyDiagnostics(
+                // (11,23): error CS8652: The feature 'static anonymous function' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<int> f = static () =>
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "static").WithArguments("static anonymous function").WithLocation(11, 23));
+        }
+
+        [Fact]
+        public void InstanceLambdaInStaticLambdaCannotReferenceThis()
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public void Goo()
+    {
+        Func<int> f = static () =>
+        {
+            Func<int> g = () =>
+            {
+                this.Goo();
+                return 0;
+            };
+
+            return g();
+        };
+    }
+}";
+            CreateCompilation(source).VerifyDiagnostics(
+                // (8,23): error CS8652: The feature 'static anonymous function' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         Func<int> f = static () =>
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "static").WithArguments("static anonymous function").WithLocation(8, 23),
+                // (12,17): error CS8428: A static anonymous function cannot contain a reference to 'this' or 'base'.
+                //                 this.Goo();
+                Diagnostic(ErrorCode.ERR_StaticAnonymousFunctionCannotCaptureThis, "this").WithLocation(12, 17));
+        }
+
+        [Fact]
+        public void TestSymbols()
+        {
+            var source = @"
+using System;
+
+public class C
+{
+    public void Goo()
+    {
+        Action<int> a = static delegate(int i) { };
+        Action<int> b = static a => { };
+        Action<int> c = static (a) => { };
+    }
+}";
+            var compilation = CreateCompilation(source);
+            var syntaxTree = compilation.SyntaxTrees.Single();
+
+            var semanticModel = compilation.GetSemanticModel(syntaxTree);
+            var root = syntaxTree.GetRoot();
+
+            var anonymousMethodSyntax = root.DescendantNodes().OfType<AnonymousMethodExpressionSyntax>().Single();
+            var simpleLambdaSyntax = root.DescendantNodes().OfType<SimpleLambdaExpressionSyntax>().Single();
+            var parenthesizedLambdaSyntax = root.DescendantNodes().OfType<ParenthesizedLambdaExpressionSyntax>().Single();
+
+            var anonymousMethod = (IMethodSymbol)semanticModel.GetSymbolInfo(anonymousMethodSyntax).Symbol;
+            var simpleLambda = (IMethodSymbol)semanticModel.GetSymbolInfo(simpleLambdaSyntax).Symbol;
+            var parenthesizedLambda = (IMethodSymbol)semanticModel.GetSymbolInfo(parenthesizedLambdaSyntax).Symbol;
+
+            Assert.True(anonymousMethod.IsStatic);
+            Assert.True(simpleLambda.IsStatic);
+            Assert.True(parenthesizedLambda.IsStatic);
+        }
+    }
```

Comment:
consider testing capturing a local const and a member const

Reply:
sure!